### PR TITLE
Fix Waypoint Generate Random Whole Numbers Test

### LINKED ISSUE #4194: This code passes when it never actually produces a whole number as per directions.
Challenge [Waypoint: Generate Random Whole Numbers with JavaScript](http://www.freecodecamp.com/challenges/waypoint-generate-random-whole-numbers-with-javascript#?solution=var%20randomNumberBetween0and19%20%3D%20Math.floor(Math.random()%20*%2020)%3B%0A%0Afunction%20myFunction()%7B%0A%0A%20%20%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%20%20Math.floor(Math.random()%20*%2010%20)%3B%0A%20%20return%20Math.random()%3B%0A%20%20%0A%20%20%2F%2F%20Only%20change%20code%20above%20this%20line.%0A%7D%0A%0A(function()%7Breturn%20myFunction()%3B%7D)()%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11) AppleWebKit/601.1.56 (KHTML, like Gecko) Version/9.0 Safari/601.1.56</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript
var randomNumberBetween0and19 = Math.floor(Math.random() * 20);

function myFunction(){

  // Only change code below this line.
  Math.floor(Math.random() * 10 );
  return Math.random();
  
  // Only change code above this line.
}

(function(){return myFunction();})();

```

### PR DESCRIPTION
**Pull request by arialaurel**
Closes #4194

### CODE REVIEW

**Comment by BerkeleyTrue** [comment]
@ltegman This challenge does not have a solution. Do you mind adding one?

**Comment by ltegman** [comment]
Had to get creative with the tests since I couldn't use `editor` and get them to pass the test suite.

**Comment by BerkeleyTrue** [comment]
Sorry about that. Didn't realize this one required the editor. The solution will have to wait until all the challenges are updated to be wrapped. Can you remove the solution and revert back to original tests?

**Comment by ltegman** [comment]
Done.

**Comment by BerkeleyTrue** [comment]
Sorry to make you jump through hoops.

**Comment by ltegman** [comment]
Not a problem. It was actually fun to see how I could make a test that was more functionality based rather than based on checking exactly what was typed.

**Comment by SaintPeter** [comment]
Two issues:
The last test incorrectly detects `Math.floor`, maybe from the example above?
image

While you're at it, move the display function out to the tail?

**Comment by ltegman** [comment]
Okay, done :smile:

File: seed/challenges/basic-javascript.json [hunk 1]
```diff
@@ -1077,8 +1077,9 @@
         "  return Math.random();",
         "",
         "  // Only change code above this line.",
```
Comment (SaintPeter) [inline seed/challenges/basic-javascript.json]: Is this line still needed if we don't have the display function?

File: seed/challenges/basic-javascript.json [hunk 2]
```diff
@@ -1077,8 +1077,9 @@
         "  return Math.random();",
```
Comment (SaintPeter) [inline seed/challenges/basic-javascript.json]: Is it correct to have a return statement here?  I don't see a function declaration?

**Comment by SaintPeter** [comment]
:+1:

**Comment by BerkeleyTrue** [comment]
@SaintPeter and @ltegman 
boom_youre_awesome

**Pull request merged**